geth/chain.py

```diff
@@ -64,44 +73,44 @@ def get_ropsten_data_dir():
     )
 
 
-def get_default_base_dir():
+def get_default_base_dir() -> str:
     return get_live_data_dir()
 
 
-def get_chain_data_dir(base_dir, name):
+def get_chain_data_dir(base_dir: str, name: str) -> str:
     data_dir = os.path.abspath(os.path.join(base_dir, name))
     ensure_path_exists(data_dir)
     return data_dir
 
 
-def get_genesis_file_path(data_dir):
+def get_genesis_file_path(data_dir: str) -> str:
     return os.path.join(data_dir, "genesis.json")
 
 
-def is_live_chain(data_dir):
+def is_live_chain(data_dir: str) -> bool:
     return is_same_path(data_dir, get_live_data_dir())
 
 
-def is_ropsten_chain(data_dir):
+def is_ropsten_chain(data_dir: str) -> bool:
     return is_same_path(data_dir, get_ropsten_data_dir())
 
 
 def write_genesis_file(
```

Comment:
`write_genesis_file` currently contains separate parameters `clique_period` and `clique_epoch`. But, the circleci output shows that the function receives a clique from `GenesisData`, I can suppose, that containing both period and epoch. And I think, we should update signature to accepting a single `clique` parameter

Like this:

```python
def write_genesis_file(
    ...
    clique: Optional[Dict[str, int]] = None,
):
    ...
    if clique is None:
        clique = {"period": 5, "epoch": 30000}
    if config is None:
        config = {
            ...
            "clique": {"period": clique.get("period"), "epoch": clique.get("epoch")},
        }
    ...
```